services/core/shared/core/compat/sdk_v5/accounts.js

```diff
@@ -58,8 +58,8 @@ const indexAccounts = async job => {
 	const { accounts } = job.data;
 	const accountsDB = await getAccountsIndex();
 	accounts.forEach(account => {
-		account.username = account.dpos.delegate.username || null;
-		account.balance = account.token.balance;
+		account.username = account.dpos ? account.dpos.delegate.username : null;
+		account.balance = account.token ? account.token.balance : null;
```

Comment:
Is there any situation where we do not get the `account.token.balance` property?